assembly/decoder.ts

```diff
@@ -196,16 +196,22 @@ export class JSONDecoder<JSONHandlerT extends JSONHandler> {
     private readString(): string {
         assert(this.readChar() == '"'.charCodeAt(0), "Expected double-quoted string");
         let savedIndex = this.state.readIndex;
-        let stringParts: Array<string> = new Array<string>();
+        let stringParts: Array<string> = null;
```

Comment:
maybe just make this a field and clear array each time instead of creating new one? Should save null checks.

Reply:
I'm trying to avoid calling "join" when it's not needed. Since most strings don't have any escaped characters, then we avoid creation of arrays at all.
null check is super cheap, like 2 gas

Reply:
got it, I see `.join` doesn't short-circuit for single element arrays :(